File: Spiel.cs
using System;
using System.Collections.Generic;
using Quizly.Enums;

namespace Quizly
{
    public class Spiel
    {
        public int SpielId { get; set; }
        public Spielmodus Modus { get; set; }
        public Spielstatus Status { get; set; }

        public Spielkonfiguration Konfiguration { get; set; }
        public Schwierigkeitsgrad Schwierigkeitsgrad { get; set; }

        public DateTime StartZeit { get; set; }
        public DateTime? EndZeit { get; set; }

        public List<Spieler> Spieler { get; set; }
        public List<Runde> Runden { get; set; }

        public void Beenden()
        {
            Status = Spielstatus.Beendet;
            EndZeit = DateTime.UtcNow;
        }
    }
}

File: Android/app/src/main/java/com/quizly/android/data/model/Spiel.kt
package com.quizly.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Ported from Spiel.cs / spiel.py. */
@Serializable
data class Spiel(
    @SerialName("spiel_id")
    val spielId: Int,
    val spielmodus: Spielmodus? = null,
    val status: Spielstatus? = null,
    @SerialName("config_id")
    val configId: Int? = null,
    @SerialName("schwierigkeitsgrad_level")
    val schwierigkeitsgradLevel: Int? = null,
    @SerialName("start_zeit")
    val startZeit: String? = null,
    @SerialName("end_zeit")
    val endZeit: String? = null,
    val spieler: List<Spieler> = emptyList(),
    val runden: List<Runde> = emptyList()
)
